playbooks/provisioning/openstack/custom-actions/add-docker-registry.yml

```diff
@@ -0,0 +1,92 @@
+---
+- hosts: cluster_hosts
+  vars:
+    registries: []
+    insecure_registries: []
+
+  tasks:
+  - name: Check if docker is even installed
+    command: docker
+    register: docker_result
+    ignore_errors: True
+
+  - name: Add additional registries if docker is available
+    when: 'docker_result | succeeded'
+    block:
+    - name: Install atomic-registries package
+      yum:
+        name: atomic-registries
+        state: latest
+
+    - name: Get registry configuration file
+      register: file_result
+      stat:
+        path: /etc/containers/registries.conf
+
+    - name: Check if it exists
+      assert:
+        that: 'file_result.stat.exists'
+        msg: "Configuration file does not exist."
+
+    - name: Load configuration file
+      shell: cat /etc/containers/registries.conf
+      register: file_content
+      become: true
+
+    - name: Store file content into a variable
+      set_fact:
+        docker_conf: "{{ file_content.stdout | from_yaml }}"
+
+    - name: Make sure that docker file content is a dictionary
+      when: '(docker_conf is string) and (not docker_conf)'
+      set_fact:
+        docker_conf: {}
+
+    - name: Make sure that registries is a list
+      when: 'registries is string'
+      set_fact:
+        registries_list: [ "{{ registries }}" ]
+
+    - name: Make sure that insecure_registries is a list
+      when: 'insecure_registries is string'
+      set_fact:
+        insecure_registries_list: [ "{{ insecure_registries }}" ]
+
+    - name: Set docker file content to a unified format
+      set_fact:
+        docker_conf_registries: "{{ [] if docker_conf['registries'] is not defined else [ docker_conf['registries'] ] if docker_conf['registries'] is string else docker_conf['registries'] }}"
```

Comment:
Have you seen `docker_conf['registries']` not being an array? When I checked the format & the source code, it seems to always expect an array.

In which case, please remove the check so the code is easier to read and understand.